Subject: [PATCH] Restore AMI DevKit wiring on the current AST2600 DTS

Port the original AMI board memory, peripheral placement and power GPIO
configuration to the current bindings. Release PWM and SPI pin selections
that conflict with the power GPIOs, retaining the current 64 MiB layout.

Upstream-Status: Inappropriate [DevKit hardware configuration]

--- a/dts-evb-ast2600/aspeed-ast2600-evb.dts
+++ b/dts-evb-ast2600/aspeed-ast2600-evb.dts
@@ -24,7 +24,7 @@
 
 	memory@80000000 {
 		device_type = "memory";
-		reg = <0x80000000 0x80000000>;
+		reg = <0x80000000 0x40000000>;
 	};
 
 	reserved-memory {
@@ -185,6 +185,7 @@
 
 	fan7: pwm-fan7 {
 		compatible = "pwm-fan";
+		status = "disabled";
 		pwms = <&pwm_tach 7 40000 0>;	/* Target freq:25 kHz */
 		cooling-min-state = <0>;
 		cooling-max-state = <3>;
@@ -194,6 +195,7 @@
 
 	fan8: pwm-fan8 {
 		compatible = "pwm-fan";
+		status = "disabled";
 		pwms = <&pwm_tach 8 40000 0>;	/* Target freq:25 kHz */
 		cooling-min-state = <0>;
 		cooling-max-state = <3>;
@@ -203,6 +205,7 @@
 
 	fan9: pwm-fan9 {
 		compatible = "pwm-fan";
+		status = "disabled";
 		pwms = <&pwm_tach 9 40000 0>;	/* Target freq:25 kHz */
 		cooling-min-state = <0>;
 		cooling-max-state = <3>;
@@ -212,6 +215,7 @@
 
 	fan10: pwm-fan10 {
 		compatible = "pwm-fan";
+		status = "disabled";
 		pwms = <&pwm_tach 10 40000 0>;	/* Target freq:25 kHz */
 		cooling-min-state = <0>;
 		cooling-max-state = <3>;
@@ -221,6 +225,7 @@
 
 	fan11: pwm-fan11 {
 		compatible = "pwm-fan";
+		status = "disabled";
 		pwms = <&pwm_tach 11 40000 0>;	/* Target freq:25 kHz */
 		cooling-min-state = <0>;
 		cooling-max-state = <3>;
@@ -270,15 +275,13 @@
 	pinctrl-0 = <&pinctrl_pwm0_default &pinctrl_pwm1_default
 		     &pinctrl_pwm2_default &pinctrl_pwm3_default
 		     &pinctrl_pwm4_default &pinctrl_pwm5_default
-		     &pinctrl_pwm6_default &pinctrl_pwm7_default
-		     &pinctrl_pwm8g1_default &pinctrl_pwm9g1_default
-		     &pinctrl_pwm10g1_default &pinctrl_pwm11g1_default
+		     &pinctrl_pwm6_default
 		     &pinctrl_pwm12g1_default &pinctrl_pwm13g1_default
 		     &pinctrl_pwm14g1_default
 		     &pinctrl_tach0_default &pinctrl_tach1_default
 		     &pinctrl_tach2_default &pinctrl_tach3_default
 		     &pinctrl_tach4_default &pinctrl_tach5_default
-		     &pinctrl_tach6_default &pinctrl_tach7_default
+		     &pinctrl_tach6_default
 		     &pinctrl_tach8_default &pinctrl_tach9_default
 		     &pinctrl_tach10_default &pinctrl_tach11_default
 		     &pinctrl_tach12_default &pinctrl_tach13_default
@@ -303,9 +306,6 @@
 	};
 	fan-6 {
 		tach-ch = /bits/ 8 <0x6>;
-	};
-	fan-7 {
-		tach-ch = /bits/ 8 <0x7>;
 	};
 	fan-8 {
 		tach-ch = /bits/ 8 <0x8>;
@@ -552,7 +552,7 @@
 &spi2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_spi2_default &pinctrl_spi2cs1_default
-				&pinctrl_spi2cs2_default &pinctrl_qspi2_default>;
+				&pinctrl_qspi2_default>;
 	status = "okay";
 
 	flash@0 {
@@ -644,6 +644,11 @@
 
 &i2c0 {
 	status = "okay";
+
+	temp@2e {
+		compatible = "adi,adt7490";
+		reg = <0x2e>;
+	};
 };
 
 &i2c1 {
@@ -700,11 +705,6 @@
 &i2c7 {
 	status = "okay";
 
-	temp@2e {
-		compatible = "adi,adt7490";
-		reg = <0x2e>;
-	};
-
 	eeprom@50 {
 		compatible = "atmel,24c08";
 		reg = <0x50>;
@@ -751,6 +751,10 @@
 };
 
 &i2c15 {
+	status = "okay";
+};
+
+&fsim0 {
 	status = "okay";
 };
 
@@ -860,7 +864,7 @@
 };
 #endif
 
-#if 0
+#if 1
 &gfx {
 	status = "okay";
 	memory-region = <&gfx_memory>;
@@ -931,7 +935,7 @@
 };
 #endif
 
-#if 0
+#if 1
 &mbox {
 	status = "okay";
 };
@@ -1190,15 +1194,15 @@
     /*M0-M7*/       "","","","","","","","",
     /*N0-N7*/       "","","","","","","","",
     /*O0-O7*/       "","","","","","","","",
-    /*P0-P7*/       "","","","","","","","",
-    /*Q0-Q7*/       "","","","","","","","",
+    /*P0-P7*/       "RESET_BUTTON","RESET_OUT","POWER_BUTTON","POWER_OUT","","","","",
+    /*Q0-Q7*/       "","","","","","","","PS_PWROK",
     /*R0-R7*/       "","","","","","","","",
     /*S0-S7*/       "","","","","","","","",
     /*T0-T7*/       "","","","","","","","",
     /*U0-U7*/       "","","","","","","","",
-    /*V0-V7*/       "NMI_OUT","RESET_BUTTON","NMI_BUTTON","POWER_BUTTON","PS_PWROK","POST_COMPLETE","POWER_OUT","RESET_OUT",
+    /*V0-V7*/       "NMI_OUT","","NMI_BUTTON","","","","","",
     /*W0-W7*/       "ID_BUTTON","SIO_S5","SIO_ONCONTROL","","","","","",
-    /*X0-X7*/       "","","","","","","","",
+    /*X0-X7*/       "","","POST_COMPLETE","","","","","",
     /*Y0-Y7*/       "","TEST_INTR","","","","","","",
     /*Z0-Z7*/       "","","","","","","","";
 };
